src/blockchains/eos/formulas.ts

```diff
@@ -0,0 +1,94 @@
+import Decimal from 'decimal.js';
+
+const ZERO = new Decimal(0);
+const ONE = new Decimal(1);
+const MAX_RATIO = new Decimal(1000000);
+const MAX_FEE = new Decimal(1000000);
+
+export function init() {
+    Decimal.set({precision: 100, rounding: Decimal.ROUND_DOWN});
+}
+
+export function calculatePurchaseReturn(supply, reserveBalance, reserveRatio, depositAmount) {
```

Comment:
They're used only by eos. The actual question to be asked is, why do we need them at all?

Reply:
You can't execute EOS functions on chain from off-chain code. But I would imagine that these functions can be generic so that we can re-use them in other blockchains that don't support it.

Reply:
OK, will move back to toplevel directory.